xmodule/modulestore/tests/test_mongo.py

```diff
@@ -787,10 +787,15 @@ def _check_write(self, key, value):
         assert self.kvs.get(key) == value
 
     def test_write(self):
-        yield (self._check_write, KeyValueStore.Key(Scope.content, None, None, 'foo'), 'new_data')
-        yield (self._check_write, KeyValueStore.Key(Scope.children, None, None, 'children'), [])
-        yield (self._check_write, KeyValueStore.Key(Scope.children, None, None, 'parent'), None)
-        yield (self._check_write, KeyValueStore.Key(Scope.settings, None, None, 'meta'), 'new_settings')
+        items = [
+            {"scope": Scope.content, "key": "foo", "expected": "new_data"},
+            {"scope": Scope.children, "key": "children", "expected": []},
+            {"scope": Scope.children, "key": "parent", "expected": None},
+            {"scope": Scope.settings, "key": "meta", "expected": "new_settings"},
+        ]
+        for item in items:
+            with self.subTest():
+                self._check_write(KeyValueStore.Key(item["scope"], None, None, item["key"]), item["expected"])
```

Comment:
We're using pytest, so subtests don't work here. Using `ddt` will be a better idea. Do do this:
1. Import `ddt`.
2. Add `@ddt.ddt` to the `TestMongoKeyValueStore` class.
3. Rewrite tests to something like this:
```python
    @ddt.data(
        (Scope.content, "foo", "new_data"),
        (Scope.children, "children", []),
    )
    @ddt.unpack
    def test_write(self, scope, key, expected):
        self._check_write(KeyValueStore.Key(scope, None, None, key), expected)
```

Reply:
Thanks for the feedback, I'll tryout this method.